app/services/experiment_control_service.rb

```diff
@@ -46,50 +46,35 @@ def start_current_patient_experiment(name, days_til_start, days_til_end, percent
     def start_inactive_patient_experiment(name, days_til_start, days_til_end)
       experiment = Experimentation::Experiment.find_by!(name: name, experiment_type: "stale_patients")
       total_days = days_til_end - days_til_start + 1
-      date = days_til_start.days.from_now.to_date
-      eligibility_start = (date - INACTIVE_PATIENTS_ELIGIBILITY_START).beginning_of_day
-      eligibility_end = (date - INACTIVE_PATIENTS_ELIGIBILITY_END).end_of_day
+      start_date = days_til_start.days.from_now.to_date
+      eligibility_start = (start_date - INACTIVE_PATIENTS_ELIGIBILITY_START).beginning_of_day
+      eligibility_end = (start_date - INACTIVE_PATIENTS_ELIGIBILITY_END).end_of_day
+      range = Range.new(eligibility_start, eligibility_end)
 
-      experiment.update!(state: "selecting", start_date: days_til_start.days.from_now.to_date, end_date: days_til_end.days.from_now.to_date)
+      experiment.update!(state: "selecting", start_date: start_date, end_date: days_til_end.days.from_now.to_date)
 
-      eligible_ids = patient_pool
-        .joins(:encounters)
-        .where(encounters: {device_created_at: eligibility_start..eligibility_end})
-        .where("NOT EXISTS (SELECT 1 FROM encounters WHERE encounters.patient_id = patients.id AND
-                encounters.device_created_at > ?)", eligibility_end)
-        .left_joins(:appointments)
-        .where("NOT EXISTS (SELECT 1 FROM appointments WHERE appointments.patient_id = patients.id AND
-                appointments.scheduled_date >= ?)", date)
-        .distinct
-        .pluck(:id)
+      eligible_ids = Experimentation::StalePatientSelection.call(start_date: start_date, eligible_range: range)
       eligible_ids.shuffle!
 
       total_days.times do
         daily_ids = eligible_ids.pop(PATIENTS_PER_DAY)
         break if daily_ids.empty?
+        schedule_date = start_date
```

Comment:
Doesn't this need to be done before the day loop begins? Won't all appointments be scheduled for the same day? This points to the need for a new spec for "schedules patients for different days of the experiment". It could stub PATIENTS_PER_DAY to 1 and confirm that 2 patients have reminders scheduled on different days.

Reply:
Good catch - fixed in b9a8120f87835b0ff5bdb86238a7d31dec28c8a7.